src/main/java/org/zalando/nakadi/enrichment/MetadataEnrichmentStrategy.java

```diff
@@ -30,7 +30,9 @@ private void setVersion(final JSONObject metadata, final EventType eventType) {
     }
 
     private void setFlowId(final JSONObject metadata) {
-        metadata.put("flow_id", FlowIdUtils.peek());
+        if (metadata.optString("flow_id") == null || "".equals(metadata.optString("flow_id"))) {
```

Comment:
optString does not return null, it can return "null", "" or "value"
https://github.com/stleary/JSON-java/blob/ef7a5e40be2e6109b31957bc56b5b80943cc1f92/JSONObject.java#L1345

and I think it makes sense to move `metadata.optString("flow_id")` to var since you use it twice.